web/api/v1/api.go

```diff
@@ -477,13 +477,93 @@ func (api *API) labelNames(r *http.Request) apiFuncResult {
 	return apiFuncResult{names, nil, warnings, nil}
 }
 
+var (
+	minTime = time.Unix(math.MinInt64/1000+62135596801, 0).UTC()
+	maxTime = time.Unix(math.MaxInt64/1000-62135596801, 999999999).UTC()
+
+	minTimeFormatted = minTime.Format(time.RFC3339Nano)
+	maxTimeFormatted = maxTime.Format(time.RFC3339Nano)
+)
+
+func parseTimeParam(r *http.Request, paramName string, defaultValue time.Time) (time.Time, error) {
```

Comment:
to speed up the reviews such changes that can be indipendent from the main PR could be opened as a separate PR

Reply:
Ok, understand. What do you suggest - to revert this back and to open another PR?

Reply:
I think it might be fine, the PR is small enough as is.

Reply:
why not return timestamps here to avoid the duplicate call to `timestamp.FromTime`

Reply:
Yes, good catch, thanks!

Reply:
From the other side returning ```Time``` is more flexible - you can take advantage of its methods but I guess opposite conversion is also possible(from ```int64``` to ```Time```)